aodncore/pipeline/handlerbase.py

```diff
@@ -1022,6 +1022,20 @@ def postprocess(self):  # pragma: no cover
     # "public" methods
     #
 
+    def download_remotepipelinefilecollection(self, remotepipelinefilecollection, local_path=None):
+        """Helper method to download a RemotePipelineFileCollection or RemotePipelineFile using the handler's internal
+            storage broker
+
+        :param remotepipelinefilecollection: RemotePipelineFileCollection to download
+        :param local_path: local path where files will be downloaded. Defaults to the handler's :attr:`temp_dir` value.
+        :return: None
+        """
+        remotepipelinefilecollection = ensure_remotepipelinefilecollection(remotepipelinefilecollection)
+        if local_path is None:
+            local_path = self.temp_dir
+
+        remotepipelinefilecollection.download(self._upload_store_runner.broker, local_path)
```

Comment:
Without interrupting the extended philosophical discussion above, I just wanted to point out that (consistent with @lwgordonimos' "internal mental model"), this line should really be
```python
    self._upload_store_runner.broker.download(remotepipelinefilecollection, local_path)
```
As a bonus, the [broker's download method](https://github.com/aodn/python-aodncore/blob/58d55681205dec471c1ad0f77e15f489230b53d3/aodncore/pipeline/storage.py#L109) also accepts either a RemotePipelineFileCollection or a single RemotePipelineFile, so you don't even need to do the `ensure_remotepipelinefilecollection` above.

Reply:
Yeap, and if RemoteFile got the same method, this function could be just named differently and accept any of the inputs.

My point of turning both classes into boring stuff (e.g. named tuples/dicts) was that, as of today, would require some effort (that for me is hard to grasp since this package is used everywhere).

Reply:
@mhidas I realised that exact thing as soon is it was merged unfortunately. In fact, I realised that the handler was the wrong place as well anyway, since we went to the trouble of making a StateQuery class to be the state API for exactly this kind of thing :/

https://github.com/aodn/python-aodncore/pull/209

@ocehugo remember that this is not a general purpose library for interacting with our data, this is a *framework* for keeping handlers as consistent as possible. This is why it is so opinionated about how things should be done in terms of the workflow during execution. 

We can implement that kind of flexible download interface without methods on those classes or dismantling other aspects of the design which are useful and/or intentional based on real world use cases we've encountered along the way (and in the previous pipeline implementation).

With the above change, this is the kind of code handlers can write to download files if they really need to, and this is the kind of style I would encourage.

```python
myfilter = SomeOgcFilter()
collection = self.state_query.query_wfs_urls_for_layer('mylayer', myfilter)
self.state_query.download(collection, self.temp_dir)

# and of course there's nothing preventing you from
# looking at/manipulating your collection to get whatever info you need, e.g.
bigger_than_a_kb = RemotePipelineFileCollection(f for f in collection if f.size > 1024)
only_netcdfs = collection.filter_by_attribute_id(FileType.NETCDF)
```